{
  "active_lock_reason": null,
  "assignee": {
    "avatar_url": "https://avatars3.githubusercontent.com/u/91730?v=4",
    "events_url": "https://api.github.com/users/cofyc/events{/privacy}",
    "followers_url": "https://api.github.com/users/cofyc/followers",
    "following_url": "https://api.github.com/users/cofyc/following{/other_user}",
    "gists_url": "https://api.github.com/users/cofyc/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/cofyc",
    "id": 91730,
    "login": "cofyc",
    "node_id": "MDQ6VXNlcjkxNzMw",
    "organizations_url": "https://api.github.com/users/cofyc/orgs",
    "received_events_url": "https://api.github.com/users/cofyc/received_events",
    "repos_url": "https://api.github.com/users/cofyc/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/cofyc/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/cofyc/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/cofyc"
  },
  "assignees": [
    {
      "avatar_url": "https://avatars3.githubusercontent.com/u/91730?v=4",
      "events_url": "https://api.github.com/users/cofyc/events{/privacy}",
      "followers_url": "https://api.github.com/users/cofyc/followers",
      "following_url": "https://api.github.com/users/cofyc/following{/other_user}",
      "gists_url": "https://api.github.com/users/cofyc/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/cofyc",
      "id": 91730,
      "login": "cofyc",
      "node_id": "MDQ6VXNlcjkxNzMw",
      "organizations_url": "https://api.github.com/users/cofyc/orgs",
      "received_events_url": "https://api.github.com/users/cofyc/received_events",
      "repos_url": "https://api.github.com/users/cofyc/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/cofyc/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/cofyc/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/cofyc"
    }
  ],
  "author_association": "CONTRIBUTOR",
  "body": "<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->\r\n\r\n\r\n**What happened**: If a PVC with delayed binding fails to provision (for example due to a StorageClass parameter error), even when the error is fixed later on, provisioning never succeeds.\r\n\r\n**What you expected to happen**: After the error is fixed, provisioning should retry and succeed.\r\n\r\n**How to reproduce it (as minimally and precisely as possible)**:\r\n- Create a StorageClass with VolumeBindingMode = WaitForFirstConsumer, and set some garbage parameters to eventually cause provisioning to fail.\r\n- Create a PVC using the StorageClass\r\n- Create a pod that consumes the PVC. Provisioning should fail\r\n- Recreate the StorageClass with parameters removed.\r\nUnder correct operation, the provisioner should successfully retry and create a PV, and the pod should proceed to start. But this is not the case.\r\n\r\n**Anything else we need to know?**:\r\nWhen the PVC first fails to provision, the consumer pod has the following events:\r\n```\r\n  Type     Reason            Age                From               Message\r\n  ----     ------            ----               ----               -------\r\n  Warning  FailedScheduling  37s                default-scheduler  selectedNode annotation value \"\" not set to scheduled node \"[node]\"\r\n  Warning  FailedScheduling  37s                default-scheduler  pod has unbound immediate PersistentVolumeClaims (repeated 2 times)\r\n  Warning  FailedScheduling  37s                default-scheduler  AssumePod failed: pod 8e98377a-fcd7-11e8-beb7-42010a800002 is in the cache, so can't be assumed\r\n  Warning  FailedScheduling  0s (x18 over 37s)  default-scheduler  pod has unbound immediate PersistentVolumeClaims (repeated 3 times)\r\n```\r\n\r\nafter the error is fixed, the last event (unbound immediate PVCs) keeps firing.\r\n\r\nFor some reason the scheduler doesn't think the PVC uses delayed binding:\r\nhttps://github.com/kubernetes/kubernetes/blob/67bc6d2083d94b38e5c8b249fe91aeafa1a0af19/pkg/controller/volume/persistentvolume/scheduler_binder.go#L504\r\n\r\nOne possible cause is somehow the scheduler thinks the selectedNode annotation still exists in the PVCCache (the provisioner removed the annotation after provisioning failure):\r\nhttps://github.com/kubernetes/kubernetes/blob/67bc6d2083d94b38e5c8b249fe91aeafa1a0af19/pkg/controller/volume/persistentvolume/pv_controller.go#L297\r\nOr the scheduler failed to get the StorageClass. I'm not sure atm why either could happen.\r\n\r\n**Environment**:\r\n- Kubernetes version (use `kubectl version`): 1.12.3, 1.13\r\n- Cloud provider or hardware configuration:\r\n- OS (e.g. from /etc/os-release):\r\n- Kernel (e.g. `uname -a`):\r\n- Install tools:\r\n- Others:\r\n\r\n/sig storage\r\n/cc @msau42 \r\n\r\n<!-- DO NOT EDIT BELOW THIS LINE -->\r\n/kind bug",
  "closed_at": "2019-01-12T01:42:32Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/20407524?v=4",
    "events_url": "https://api.github.com/users/k8s-ci-robot/events{/privacy}",
    "followers_url": "https://api.github.com/users/k8s-ci-robot/followers",
    "following_url": "https://api.github.com/users/k8s-ci-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/k8s-ci-robot/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/k8s-ci-robot",
    "id": 20407524,
    "login": "k8s-ci-robot",
    "node_id": "MDQ6VXNlcjIwNDA3NTI0",
    "organizations_url": "https://api.github.com/users/k8s-ci-robot/orgs",
    "received_events_url": "https://api.github.com/users/k8s-ci-robot/received_events",
    "repos_url": "https://api.github.com/users/k8s-ci-robot/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/k8s-ci-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/k8s-ci-robot/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/k8s-ci-robot"
  },
  "comments": 12,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/71928/comments",
  "created_at": "2018-12-11T00:38:53Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/71928/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/71928",
  "id": 389547887,
  "labels": [
    {
      "color": "e11d21",
      "default": false,
      "description": "Categorizes issue or PR as related to a bug.",
      "id": 105146071,
      "name": "kind/bug",
      "node_id": "MDU6TGFiZWwxMDUxNDYwNzE=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/bug"
    },
    {
      "color": "d2b48c",
      "default": false,
      "description": "Categorizes an issue or PR as relevant to SIG Storage.",
      "id": 169428334,
      "name": "sig/storage",
      "node_id": "MDU6TGFiZWwxNjk0MjgzMzQ=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/sig/storage"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/71928/labels{/name}",
  "locked": false,
  "milestone": null,
  "node_id": "MDU6SXNzdWUzODk1NDc4ODc=",
  "number": 71928,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "When a PVC is provisioned with delayed binding, it never recovers from a failed provision",
  "updated_at": "2019-07-10T11:38:13Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/71928",
  "user": {
    "avatar_url": "https://avatars1.githubusercontent.com/u/1689416?v=4",
    "events_url": "https://api.github.com/users/verult/events{/privacy}",
    "followers_url": "https://api.github.com/users/verult/followers",
    "following_url": "https://api.github.com/users/verult/following{/other_user}",
    "gists_url": "https://api.github.com/users/verult/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/verult",
    "id": 1689416,
    "login": "verult",
    "node_id": "MDQ6VXNlcjE2ODk0MTY=",
    "organizations_url": "https://api.github.com/users/verult/orgs",
    "received_events_url": "https://api.github.com/users/verult/received_events",
    "repos_url": "https://api.github.com/users/verult/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/verult/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/verult/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/verult"
  }
}